src/server/game/Mails/ServerMailMgr.cpp

```diff
@@ -0,0 +1,331 @@
+/*
+ * This file is part of the AzerothCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by the
+ * Free Software Foundation; either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ServerMailMgr.h"
+#include "AchievementMgr.h"
+#include "DatabaseEnv.h"
+#include "Item.h"
+#include "Log.h"
+#include "ObjectMgr.h"
+#include "Player.h"
+#include "QuestDef.h"
+#include "SharedDefines.h"
+#include "Timer.h"
+
+ServerMailMgr::ServerMailMgr() {}
+ServerMailMgr::~ServerMailMgr() {}
+
+ServerMailMgr* ServerMailMgr::instance()
+{
+    static ServerMailMgr instance;
+    return &instance;
+}
+
+void ServerMailMgr::LoadMailServerTemplates()
+{
+    uint32 oldMSTime = getMSTime();
+
+    _serverMailStore.clear(); // for reload case
+
+    //                                                    0     1         2         3          4       5
+    QueryResult result = CharacterDatabase.Query("SELECT `id`, `moneyA`, `moneyH`, `subject`, `body`, `active` FROM `mail_server_template`");
+    if (!result)
+    {
+        LOG_INFO("server.loading", ">> Loaded 0 server mail rewards. DB table `mail_server_template` is empty.");
+        LOG_INFO("server.loading", " ");
+        return;
+    }
+
+    _serverMailStore.rehash(result->GetRowCount());
+
+    do
+    {
+        Field* fields = result->Fetch();
+
+        uint32 id = fields[0].Get<uint32>();
+
+        ServerMail& servMail = _serverMailStore[id];
+
+        servMail.id          = id;
+        servMail.moneyA      = fields[1].Get<uint32>();
+        servMail.moneyH      = fields[2].Get<uint32>();
+        servMail.subject     = fields[3].Get<std::string>();
+        servMail.body        = fields[4].Get<std::string>();
+        servMail.active      = fields[5].Get<uint8>();
+
+        if (servMail.moneyA > MAX_MONEY_AMOUNT || servMail.moneyH > MAX_MONEY_AMOUNT)
+        {
+            LOG_ERROR("sql.sql", "Table `mail_server_template` has moneyA {} or moneyH {} larger than MAX_MONEY_AMOUNT {} for id {}, skipped.", servMail.moneyA, servMail.moneyH, MAX_MONEY_AMOUNT, servMail.id);
+            return;
+        }
+    } while (result->NextRow());
+
+    LoadMailServerTemplatesItems();
+    LoadMailServerTemplatesConditions();
```

Comment:
Note to self:
Should probably pass templateID into these and only load them if we load the template. As of now we always load these no matter if the checks above was made or not

Reply:
actually already do it